File: Day7/part1.py
def part1():
    with open("Day7/input.txt", "r") as f:
        input = f.read().splitlines()

    beams = set()
    beams.add(input[0].find("S"))

    score = 0
    for row in range(2,len(input)):
        to_remove = set()
        to_add = set()
        for pos in beams:
            if input[row][pos] == "^":
                score+=1
                to_remove.add(pos)
                to_add.add(max(0,pos-1))
                to_add.add(min(len(input[0])-1, pos+1))

        for pos in to_remove:
            beams.remove(pos)

        for pos in to_add:
            beams.add(pos)

    print(score)



if __name__ == "__main__":
    part1()

File: Day7/part1.php
<?php
namespace Aoc2025\Day7;

function part1(){
    $input = file("Day7/example.txt", FILE_IGNORE_NEW_LINES) or die("File not found");

    $srcPos = strpos($input[0], "S");
    $beamsPos = [$srcPos];
    $maxPos = strlen($input[0])-1;

    $score = 0;
    for ($row = 2; $row < count($input); $row++){
        $toRemove = [];
        $toAdd = [];
        print_r($beamsPos);
        foreach($beamsPos as $i => $pos){
            if ($input[$row][$pos]=== "^"){
                $score++;
                $toAdd[] =max(0,$pos-1); 
                $toAdd[] =min($maxPos, $pos+1); 
                $toRemove[] = $pos;
            }
        }

        foreach($toRemove as $pos){
            array_splice($beamsPos, array_search($pos, $beamsPos),1);
        }

        foreach($toAdd as $pos){
            if(!in_array($pos, $beamsPos)){
                $beamsPos[] = $pos;
            }
        }
    }

    print_r("$score \n");
}
